### DISCUSSION TITLE
Cannot see mouse pointer

### DISCUSSION THREAD

**Discussion by Duzeper** [rern/rAudio#1738] [Bug Reports]
Have just installed rAudio 20240615 on a RPi 3B+ but I have no idea if it is any good because the mouse pointer is invisible. I have to, aimlessly wander around the screen until somethong lights up. I managed to get the Restart/Shutdown buttons, but after 15 minutes of wandering around I just gave up and powered off. No way of knowing what this may have done to the system til I try again to boot it, although that won't be happening unless someone can tell me how to fix it.
I am using a wired USB mouse and keyboard as experience tells me that I need to start with that in order to configure the Bluetooth devices, assuming it is possible to do so.

**Comment by MulberryPi** [comment]
> Have just installed rAudio 20240615 on a RPi 3B+ but I have no idea if it is any good because the mouse pointer is invisible. I have to, aimlessly wander around the screen until somethong lights up. I managed to get the Restart/Shutdown buttons, but after 15 minutes of wandering around I just gave up and powered off. No way of knowing what this may have done to the system til I try again to boot it, although that won't be happening unless someone can tell me how to fix it. I am using a wired USB mouse and keyboard as experience tells me that I need to start with that in order to configure the Bluetooth devices, assuming it is possible to do so.

I experienced this issue by accident with 20240615 on Raspberry Pi 3 when I unplugged my wireless keyboard/mouse USB dongle from the system while the system was running/powered on. After that, nothing I could do could bring the mouse cursor back, so I reimaged the SD card and started again and this time made sure the USB dongle was always plugged in and never "hot swapped"

If you reimage the SD card then boot up again with the USB mouse keyboard plugged in, does this fix the issue?

**Reply by MulberryPi** [reply]
Also, if you are only needing the local browser to perform initial configuration for bluetooth etc, you can instead perform the initial setup on the headless system by connecting to the rAudio hotspot (SSID: rAudio PW: raudioap) after booting. then once connected to hotspot you can configure whatever network/bluetooth setting you need (via browser to 192.168.5.1) then disconnect from the hotspot.

**Comment by Duzeper** [comment]
I have now tried that but it has had no effect.
Frankly, I am not sure how it could make a difference anyway, by reimaging the SD card on a PC, transferring it to the RPi with USB mouse and keyboard already connected and then rebooting. Surely, the reimaging process 'knows nothing' about the state of the RPi it is going to run on.
Also, Re. your hotspot comment, firstly, I am pretty much a noob when it comes to the RPi, so I am not sure I understand quite what you are saying there, and even if it was possible to set up my bluetooth mouse/keyboard, I wonder if that would actually fix the pointer issue?
I have just, by a lot of trial and error, managed to reach the 'Networks' page which, on the face of it, looks as though that is where I can set up my bluetooth devices but, no matter how much I move the mouse around, I cannot  get it to highlight the bluetooth button, and now, I cannot get away from it.
It does seem that it is possible to access the buttons at the bottom of the screen and the settings drop-down but nothing else, including Restart/Shutdown.

**Reply by MulberryPi** [reply]
It made a difference in my case because whatever happened when I unplugged the USB dongle with system powered up broke something in the OS which a reimage/factory reset seemed to have fixed. Your circumstance may be slightly different. 

It sounds to me like you are using a display and mouse/keyboard connected directly to the Pi, but many including myself control rAudio via a web browser over local area network and have the Pi headless (headless meaning no display or keyboard/mouse attached to the Pi)

Once you image the SD card and load it up on the Pi, you should actually see a new wireless network appear called rAudio, if you connect to this via the credentials i listed above, then type in 192.168.5.1 on your browser URL, you should then see rAudio interface. From there you can set up your bluetooth, WLAN, etc, then once you save those settings you can then disconnect from the hotspot and the system should now be connected to your LAN and be given an IP by your network to which you now use to access rAudio. This method allows you to get everything up and running.

Obviously if you don't wish to use the above and do wish to use a direct connected display/mouse etc then your point stands, but I don't have any further advice on why the cursor has disappeared sorry.

**Comment by Duzeper** [comment]
OK, so that worked, in that I could see the app on my tablet. I tried to set up the bluetooth devices and although it claimed to have paired with the mouse, it didn't work, and the keyboard pairing failed. 
Then I connected the HDMI monitor while connected to the tablet by wifi then the monitor resolution dropped way down so I couldn't see all the controls, although there was a pointer visible and the wired mouse worked OK
Next I disconnected the tablet and rebooted the RPi as before and the mouse has disappeared again. Clearly this app only works headless on a touch screen, which is not what I want. I'll give it some thought but I suspect this is not for me.

Thanks again for your help

**Reply by MulberryPi** [reply]
My main lounge rAudio system always has a monitor connected via HDMI, and logitech K400 wireless keyboard, the mouse cursor always works correctly and only once it broke when i unplug the USB dongle while powered on, so the system IS designed to work "headed", it seems you are having some strange issues unfortunately, Whatever is responsible for rendering the mouse cursor sprite is playing up on your system.

i20240818 image has just been released, it might be worth downloading this to give it a try?

Sorry couldn't be of more help, I'm new to this myself and don't fully understand the nuts and bolts behind the software. Good luck!

**Comment by Duzeper** [comment]
Thanks for that suggestion. The new image worked! Unfortunately it is a case of 1 step forwards and 2 steps back. So I now have a working wired mouse and keyboard, Of my 2 bluetooth keyboards, I can only get 1 to work. The other is not even discovered. It's not the keyboard I wanted to use, but it will do the job. Unfortunately, although it will discover the bluetooth mouse, pairing consistently fails. In fact I can now see 2 versions of the mouse ('cos I reset the mouse and tried again and it came up as another mouse, which also will not pair) and I can't find any way to remove them.
In a vain attempt to see if there was any other way, I clicked on the 'System' button which resulted in the screen displaying a flashing rAudio icon. As I write this, it has been there for about an hour, so it clearly is never going and, of course, the RPi has effectively crashed.

Thanks for your help, but I think I am going to give this app a miss. It doesn't seem to have been thought through very well.

**Comment by ceejayemm** [comment]
@Duzeper 

Ok, lets wind things back a bit.

First things first.  rAudio is really meant to be used from a remote web browser session on a separate PC/Mac or an app on a mobile phone. It can be used with a local mouse/keyboard but this, as you are finding, is not quite so easy - forget those for the time being, in fact I would disconnect them entirely.

You have  obviously flashed an SD card with an rAudio image file.  I would suggest that if this isn't the latest version (i20240818) then you reflash the card with this version.  Are you going to be using a wired or wireless connection ?  If you are using a wired connection make sure the network cable is connected to the RPi.  If you plan on using a wireless connection you will need to add another card to the BOOT  partition of the SD card before you place it in the RPi - see #364 , my comment 5 June 2021 for the details.

Boot the RPi from the SD card (with the additional file if using wireless) and wait for the boot process to complete.  Then, on a separate PC/Mac on the same network, use a web browser to access http://raudio.local.  This should open the rAudio web page in the browser and you can then do all the necessary configuration from there.  Some users do use SSH to access the command line on the RPi but I very rarely have to do this.  You should be able to get rAudio working to your satisfaction from the web browser. 

Once you have the basic setup complete and working then you may want to branch out with other setups (DACs, HATS, access points, local browser, keyboard etc etc) but start simple and expand as you knowledge increases. rAudio is really great once you get used to it but it sometimes can be a bit daunting to get going. Once you a basic system then myself and others are usually willing to help out if we can.

Best wishes

Chris
